src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowRecordConverter.java

```
@@ -139,4 +165,46 @@ public static Boolean convertToBooleanValue(String fieldValue) {
       String.format("Field with value '%s' is not in valid format.", fieldValue));
     
   }
+
+  @VisibleForTesting
+  public static LocalDateTime parseDateTimeFormat(String fieldValue, String fieldName, String displayName) {
+    for (DateTimeFormatter dateTimeFormatter : DATE_TIME_FORMATTER) {
+      try {
+        return LocalDateTime.parse(fieldValue, dateTimeFormatter);
+      } catch (DateTimeParseException e) {
+
+      }
+    }
+    throw new UnexpectedFormatException(
+      String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+        fieldName, displayName, fieldValue));
+  }
+
+  @VisibleForTesting
+  public static LocalDate parseDateFormat(String fieldValue, String fieldName, String displayName) {
+    for (DateTimeFormatter dateFormatter : DATE_FORMATTER) {
+      try {
+        return LocalDate.parse(fieldValue, dateFormatter);
+      } catch (DateTimeParseException e) {
+
+      }
+    }
+    throw new UnexpectedFormatException(
+      String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+        fieldName, displayName, fieldValue));
+  }
+
+  @VisibleForTesting
+  public static LocalTime parseTimeFormat(String fieldValue, String fieldName, String displayName) {
+    for (DateTimeFormatter timeFormatter : TIME_FORMATTER) {
+      try {
+        return LocalTime.parse(fieldValue, timeFormatter);
+      } catch (DateTimeParseException e) {
+
```

add a comment why this exception is ignored

---

done